contracts/src/CrossChainApplications/NativeTokenBridge/NativeTokenSource.sol

```diff
@@ -144,4 +162,40 @@ contract NativeTokenSource is
             teleporterMessageID: messageID
         });
     }
+
+    /**
+     * @dev Unlocks tokens to recipient.
+     */
+    function _unlockTokens(address recipient, uint256 amount) private {
+        require(
+            recipient != address(0),
+            "ERC20TokenSource: zero recipient address"
+        );
+
+        // Transfer to recipient
+        payable(recipient).transfer(amount);
+
+        emit UnlockTokens(recipient, amount);
+    }
+
+    /**
+     * @dev Sends tokens to BLACKHOLE_ADDRESS.
+     */
+    function _burnTokens(uint256 amount) private {
+        payable(BURNED_TX_FEES_ADDRESS).transfer(amount);
+        emit BurnTokens(amount);
+    }
+
+    /**
+     * @dev Update destinationChainBurnedBalance sent from destination chain
+     */
+    function _updateDestinationChainBurnedBalance(
+        uint256 newBurnBalance
+    ) private {
+        if (newBurnBalance > destinationChainBurnedBalance) {
```

Comment:
Typing out some thoughts to make sure my understanding is correct. 

Would we want to revert in this case rather than no-op? If we revert, then the Teleporter message may be executed in the future with a call to `retryMessageExecution`. I don't think there's a scenario in which a future call would ever succeed since `destinationChainBurnedBalance` never decreases. 

With the current implementation, if the condition is not met then the Teleporter message will be considered successfully executed and the message cannot be retried in the future, which is the same outcome as the above.

I don't have a strong preference either way, but can you leave a comment explaining this behavior?

Reply:
I think it's best to have a no-op. If we revert, then `Teleporter` has to save the tx information to retry, but it will never succeed.